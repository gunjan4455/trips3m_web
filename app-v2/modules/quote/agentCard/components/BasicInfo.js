import React from 'react';
import PropTypes from 'prop-types';

import { GQuotePriceDestinationDiv } from '../G';
import { DownloadBlack } from 'helpers/Icon/Icon';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import { INFO_ROSTER } from 'constants/rosterTypes';
import { isTripConverted } from 'helpers/utils';
import QuotePrice from 'modules/quote/shared/price/QuotePrice';
import DiscountOffer from 'modules/discountEngine';

const BasicInfo = ({
                     tripId, quoteId, destination, tripStatus, priceInformation,
                     discountInformation, downloadQuote, showRoster, trackSegment
                   }) => {
  const handleDownloadQuote = () => {
    downloadQuote({ quoteId }).then((res) => {
      if (res.success) {
        window.open(res.download_url, 'target=_blank');
      } else {
        showRoster({ type: INFO_ROSTER, message: res.msg });
      }
    }).catch((ex) => {
      console.log('ex........', ex);
    });
    trackSegment(ORGANISM_CLICKED, 'Agent', '', 'Download');
  };

  const canCheckCoupons = !isTripConverted(tripStatus);
  const hasDiscountApplied = discountInformation.discount > 0;

  return (<div className="sbcw">
    <GQuotePriceDestinationDiv className="p15 flex alignCenter spaceBetween at_quoteBasicInfo">
      <p className="f14 m0 mr5 flexFull at_quoteDestination">{destination}</p>
      <div className="flexFull quotePriceInfo">
        <QuotePrice {...priceInformation} {...discountInformation} />
        <p className='pfc4 f12 text-right mt3'>Excl. of Convenience Fee</p>
      </div>
    </GQuotePriceDestinationDiv>
    <div className="p15 flex spaceBetween alignCenter">
      <button onClick={handleDownloadQuote}
              className="f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA">
        <span className="input-24 iblock mr5"><DownloadBlack /></span>
        <span>Download Quote</span>
      </button>
      {
        canCheckCoupons && !hasDiscountApplied
          ? <DiscountOffer discountInformation={discountInformation}
                           tripId={tripId} />
          : null
      }
    </div>
  </div>);
};

BasicInfo.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  destination: PropTypes.string.isRequired,
  tripStatus: PropTypes.string.isRequired,
  priceInformation: PropTypes.object.isRequired,
  discountInformation: PropTypes.object,
  downloadQuote: PropTypes.func.isRequired,
  showRoster: PropTypes.func,
  trackSegment: PropTypes.func
};

BasicInfo.defaultProps = {
  discountInformation: {},
  showRoster: () => {},
  trackSegment: () => {}
};

export default BasicInfo;
