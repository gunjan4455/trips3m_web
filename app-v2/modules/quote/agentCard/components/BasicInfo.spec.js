import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import BasicInfo from './BasicInfo';

configure({ adapter: new Adapter() });
let component;

const quoteInfoData = {
  tripId: 3918699,
  destination: "Goa",
  priceInformation: {
    price: 7499,
    priceInRupee: 7499,
    currency: 'Rupee',
    perPerson: 'per person'
  },
  downloadUrl: 'https://bookings-qa1.ttdev.in/quotes/7124217/download_quote.pdf',
  discountInformation: {
    discount: 0,
    discountableTypeId: null,
    discountableType: null
  },
  trackSegment: jest.fn,
  tripStatus: 'cancelled'
};

const createBasicInfoComponent = () => {
  return (<BasicInfo {...quoteInfoData} />);
};

describe('Quote Basic info: Render Correctly ', () => {

  beforeAll(() => {
    const BasicInfoComponent = createBasicInfoComponent();
    component = shallow(BasicInfoComponent);
  });

  it('should match with the previous snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
