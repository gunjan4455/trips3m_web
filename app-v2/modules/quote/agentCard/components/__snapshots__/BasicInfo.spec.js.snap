// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Basic info: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BasicInfo
    destination="Goa"
    discountInformation={
      Object {
        "discount": 0,
        "discountableType": null,
        "discountableTypeId": null,
      }
    }
    downloadUrl="https://bookings-qa1.ttdev.in/quotes/7124217/download_quote.pdf"
    priceInformation={
      Object {
        "currency": "Rupee",
        "perPerson": "per person",
        "price": 7499,
        "priceInRupee": 7499,
      }
    }
    showRoster={[Function]}
    trackSegment={[Function]}
    tripId={3918699}
    tripStatus="cancelled"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(render)
          className="p15 flex alignCenter spaceBetween at_quoteBasicInfo"
        >
          <p
            className="f14 m0 mr5 flexFull at_quoteDestination"
          >
            Goa
          </p>
          <div
            className="flexFull quotePriceInfo"
          >
            <QuotePrice
              currency="Rupee"
              discount={0}
              discountableType={null}
              discountableTypeId={null}
              lineBreak={false}
              perPerson="per person"
              price={7499}
              priceInRupee={7499}
              showOriginalPrice={true}
            />
          </div>
        </ForwardRef(render)>,
        <div
          className="p15 flex spaceBetween alignCenter"
        >
          <button
            className="f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA"
            onClick={[Function]}
          >
            <span
              className="input-24 iblock mr5"
            >
              <IconSvg />
            </span>
            <span>
              Download Quote
            </span>
          </button>
          <withRouter(Connect(DiscountOffer))
            discountInformation={
              Object {
                "discount": 0,
                "discountableType": null,
                "discountableTypeId": null,
              }
            }
            tripId={3918699}
          />
        </div>,
      ],
      "className": "sbcw",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p
              className="f14 m0 mr5 flexFull at_quoteDestination"
            >
              Goa
            </p>,
            <div
              className="flexFull quotePriceInfo"
            >
              <QuotePrice
                currency="Rupee"
                discount={0}
                discountableType={null}
                discountableTypeId={null}
                lineBreak={false}
                perPerson="per person"
                price={7499}
                priceInRupee={7499}
                showOriginalPrice={true}
              />
            </div>,
          ],
          "className": "p15 flex alignCenter spaceBetween at_quoteBasicInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Goa",
              "className": "f14 m0 mr5 flexFull at_quoteDestination",
            },
            "ref": null,
            "rendered": "Goa",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <QuotePrice
                currency="Rupee"
                discount={0}
                discountableType={null}
                discountableTypeId={null}
                lineBreak={false}
                perPerson="per person"
                price={7499}
                priceInRupee={7499}
                showOriginalPrice={true}
              />,
              "className": "flexFull quotePriceInfo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "Rupee",
                "discount": 0,
                "discountableType": null,
                "discountableTypeId": null,
                "lineBreak": false,
                "perPerson": "per person",
                "price": 7499,
                "priceInRupee": 7499,
                "showOriginalPrice": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "& .quotePriceInfo": Object {
                "minWidth": "200px",
                "textAlign": "right",
              },
              "borderBottom": "1px solid #EFEFEF",
              "borderTop": "1px solid #EFEFEF",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA"
              onClick={[Function]}
            >
              <span
                className="input-24 iblock mr5"
              >
                <IconSvg />
              </span>
              <span>
                Download Quote
              </span>
            </button>,
            <withRouter(Connect(DiscountOffer))
              discountInformation={
                Object {
                  "discount": 0,
                  "discountableType": null,
                  "discountableTypeId": null,
                }
              }
              tripId={3918699}
            />,
          ],
          "className": "p15 flex spaceBetween alignCenter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="input-24 iblock mr5"
                >
                  <IconSvg />
                </span>,
                <span>
                  Download Quote
                </span>,
              ],
              "className": "f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg />,
                  "className": "input-24 iblock mr5",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Download Quote",
                },
                "ref": null,
                "rendered": "Download Quote",
                "type": "span",
              },
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "discountInformation": Object {
                "discount": 0,
                "discountableType": null,
                "discountableTypeId": null,
              },
              "tripId": 3918699,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(render)
            className="p15 flex alignCenter spaceBetween at_quoteBasicInfo"
          >
            <p
              className="f14 m0 mr5 flexFull at_quoteDestination"
            >
              Goa
            </p>
            <div
              className="flexFull quotePriceInfo"
            >
              <QuotePrice
                currency="Rupee"
                discount={0}
                discountableType={null}
                discountableTypeId={null}
                lineBreak={false}
                perPerson="per person"
                price={7499}
                priceInRupee={7499}
                showOriginalPrice={true}
              />
            </div>
          </ForwardRef(render)>,
          <div
            className="p15 flex spaceBetween alignCenter"
          >
            <button
              className="f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA"
              onClick={[Function]}
            >
              <span
                className="input-24 iblock mr5"
              >
                <IconSvg />
              </span>
              <span>
                Download Quote
              </span>
            </button>
            <withRouter(Connect(DiscountOffer))
              discountInformation={
                Object {
                  "discount": 0,
                  "discountableType": null,
                  "discountableTypeId": null,
                }
              }
              tripId={3918699}
            />
          </div>,
        ],
        "className": "sbcw",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p
                className="f14 m0 mr5 flexFull at_quoteDestination"
              >
                Goa
              </p>,
              <div
                className="flexFull quotePriceInfo"
              >
                <QuotePrice
                  currency="Rupee"
                  discount={0}
                  discountableType={null}
                  discountableTypeId={null}
                  lineBreak={false}
                  perPerson="per person"
                  price={7499}
                  priceInRupee={7499}
                  showOriginalPrice={true}
                />
              </div>,
            ],
            "className": "p15 flex alignCenter spaceBetween at_quoteBasicInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Goa",
                "className": "f14 m0 mr5 flexFull at_quoteDestination",
              },
              "ref": null,
              "rendered": "Goa",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <QuotePrice
                  currency="Rupee"
                  discount={0}
                  discountableType={null}
                  discountableTypeId={null}
                  lineBreak={false}
                  perPerson="per person"
                  price={7499}
                  priceInRupee={7499}
                  showOriginalPrice={true}
                />,
                "className": "flexFull quotePriceInfo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "currency": "Rupee",
                  "discount": 0,
                  "discountableType": null,
                  "discountableTypeId": null,
                  "lineBreak": false,
                  "perPerson": "per person",
                  "price": 7499,
                  "priceInRupee": 7499,
                  "showOriginalPrice": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              Object {
                "& .quotePriceInfo": Object {
                  "minWidth": "200px",
                  "textAlign": "right",
                },
                "borderBottom": "1px solid #EFEFEF",
                "borderTop": "1px solid #EFEFEF",
              },
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA"
                onClick={[Function]}
              >
                <span
                  className="input-24 iblock mr5"
                >
                  <IconSvg />
                </span>
                <span>
                  Download Quote
                </span>
              </button>,
              <withRouter(Connect(DiscountOffer))
                discountInformation={
                  Object {
                    "discount": 0,
                    "discountableType": null,
                    "discountableTypeId": null,
                  }
                }
                tripId={3918699}
              />,
            ],
            "className": "p15 flex spaceBetween alignCenter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="input-24 iblock mr5"
                  >
                    <IconSvg />
                  </span>,
                  <span>
                    Download Quote
                  </span>,
                ],
                "className": "f14 fw7 sfc1 m0 mr5 flex alignCenter at_downloadQuoteCTA",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <IconSvg />,
                    "className": "input-24 iblock mr5",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Download Quote",
                  },
                  "ref": null,
                  "rendered": "Download Quote",
                  "type": "span",
                },
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "discountInformation": Object {
                  "discount": 0,
                  "discountableType": null,
                  "discountableTypeId": null,
                },
                "tripId": 3918699,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
